ITGlueAPI/Resources/Configurations.ps1

```diff
@@ -33,7 +68,125 @@ function New-ITGlueConfigurations {
     return $data
 }
 
+
+
 function Get-ITGlueConfigurations {
+<#
+    .SYNOPSIS
+        List all configurations in an account or organization
+
+    .DESCRIPTION
+        The Get-ITGlueConfigurations cmdlet lists all configurations
+        in an account or organization
+
+        This function can call the following endpoints:
+            Index = /configurations
+                    /organizations/:organization_id/relationships/configurations
+
+            Show =  /configurations/:id
+                    /organizations/:organization_id/relationships/configurations/:id
+
+    .PARAMETER id
+        A valid configuration Id
+
+    .PARAMETER filter_archived
+        Filter for archived
+
+        Allowed values:
+        'true', 'false', '1', '0'
+
+    .PARAMETER organization_id
+        A valid organization Id in your account
+
+    .PARAMETER filter_id
+        Filter by configuration id
+
+    .PARAMETER filter_name
+        Filter by configuration name
+
+    .PARAMETER filter_organization_id
+        Filter by organization name
+
+    .PARAMETER filter_configuration_type_id
+        Filter by configuration type id
+
+    .PARAMETER filter_configuration_status_id
+        Filter by configuration status id
+
+    .PARAMETER filter_contact_id
+        Filter by contact id
+
+    .PARAMETER filter_serial_number
+        Filter by a configurations serial number
+
+    .PARAMETER filter_psa_id
+        Filter by a PSA id
+
+    .PARAMETER filter_psa_integration_type
```

Comment:
Interesting, I was today years old when I found out this was possible. 

Overall I feel that it does have its use with small modules, functions, and really just any PoSH script that doesn't have a lot of parameters\parameterSets.  In fact, I might actually use this for some of the smaller helper functions I've written. However, when it comes to larger modules like this one it feels more natural\cleaner to keep the help section separate from the parameter section.

I did some quick looking at some of the more popular modules on GitHub and even some of the MS ones and I haven't found any other PoSH module that implements in-line help yet.

You do make a good point about how this could be a good way of encouraging others to document their parameters as they add them. Something to note though is how the parameter section might look if some of the descriptions run a little longer or are multi-lined. If everything is kept super short and to a single line then it feels like it wouldn't cause any issues.

- When it comes to `Get-Help` they both display the descriptions to the user in the same way.
- With either option of deploying about_help, the Pester `HelpComment` test would help validate any missing descriptions. 

I drafted a quick example of what it could look like below. 

```powershell
function Test-InLineHelp {

<#
.SYNOPSIS
    .
.DESCRIPTION
    .
.PARAMETER Path
    The path to the .
.PARAMETER LiteralPath
    Specifies a literal path to one or more locations.
#>

    Param(
        [String]$Path = "Help",

        # This is a simple sentence about this thing but here is also an FYI
        # You should also do this other thing because stuff
        [String]$SimpleParam = "Help",

        <#
            This is my NEW target
            This is another line to the target
            THIS WORKS to New Targets
        #>
        [String]$NewTargets = "Help",

        [String]$TestTargets = "Help",
        <#
            This is my TESTtarget
            This is another line to the target
            THIS WILL GET ASSOCIATED TO AnotherTarget
        #>

        [String]$AnotherTarget = "Help"
            #This is my ANOTHER target
            #This is another line to the target
            #TEST Target and Another Target help get combined

    )

}
```

```
NAME
    Test-InLineHelp
    
SYNOPSIS
    .
    
    
SYNTAX
    Test-InLineHelp [[-Path] <String>] [[-SimpleParam] <String>] [[-NewTargets] <String>] [[-TestTargets] <String>] [[-AnotherTarget] <String>] [<CommonParameters>]
    
    
DESCRIPTION
    .


PARAMETERS
    -Path <String>
        The path to the .

        Required?                    false
        Position?                    1
        Default value                Help
        Accept pipeline input?       false
        Accept wildcard characters?  false

    -SimpleParam <String>
        This is a simple sentence about this thing but here is also an FYI
        You should also do this other thing because stuff

        Required?                    false
        Position?                    2
        Default value                Help
        Accept pipeline input?       false
        Accept wildcard characters?  false

    -NewTargets <String>
        This is my NEW target
        This is another line to the target
        THIS WORKS to New Targets

        Required?                    false
        Position?                    3
        Default value                Help
        Accept pipeline input?       false
        Accept wildcard characters?  false

    -TestTargets <String>

        Required?                    false
        Position?                    4
        Default value                Help
        Accept pipeline input?       false
        Accept wildcard characters?  false

    -AnotherTarget <String>
        This is my TESTtarget
        This is another line to the target
        THIS WILL GET ASSOCIATED TO AnotherTarget

        This is my ANOTHER target
        This is another line to the target
        TEST Target and Another Target help get combined

        Required?                    false
        Position?                    5
        Default value                Help
        Accept pipeline input?       false
        Accept wildcard characters?  false

    <CommonParameters>
        This cmdlet supports the common parameters: Verbose, Debug,
        ErrorAction, ErrorVariable, WarningAction, WarningVariable,
        OutBuffer, PipelineVariable, and OutVariable. For more information, see
        about_CommonParameters (https:/go.microsoft.com/fwlink/?LinkID=113216).

INPUTS
OUTPUTS
RELATED LINKS
```

Reply:
Seeing it written out like that really makes it easy to lose sight of the parameter section. I'm down to roll with having everything in one place instead of splitting it up!